lib/libxdp/xsk.c

```diff
@@ -418,10 +451,13 @@ static int xsk_get_max_queues(char *ifname)
 	}
 
 	if (err) {
-		/* If the device says it has no channels, then all traffic
+		/* If the device says it has no channels, 
+		 * try to get rx tx from sysfs, otherwise all traffic
 		 * is sent to a single stream, so max queues = 1.
 		 */
-		ret = 1;
+		__u32 rx = 0, tx = 0;
```

Comment:
Not necessary to initialize these since you do this in xsk_get_queues_from_sysfs(). __u32 rx, tx; is enough.